packages/contract-dependencies-ethers/src/ContractDependenciesEthers.ts

```diff
@@ -83,13 +117,39 @@ export class ContractDependenciesEthers implements Dependencies<BigNumber> {
         return <string>this.address;
     }
 
+    public registerTransactionStatusCallback(key: string, callback: TransactionStatusCallback): void {
+        this.transactionStatusCallbacks[key] = callback;
+    }
+
+    public deRegisterTransactionStatusCallback(key: string): void {
+        delete this.transactionStatusCallbacks[key];
+    }
+
+    public deRegisterAllTransactionStatusCallbacks(): void {
+        Object.keys(this.transactionStatusCallbacks).map((key) => this.deRegisterTransactionStatusCallback(key));
+    }
+
+    public onTransactionStatusChanged(txMetadata: TransactionMetadata, status: TransactionStatus, hash?: string): void {
+        for (let callback of Object.values(this.transactionStatusCallbacks)) {
+            callback(txMetadata, status, hash);
+        }
+    }
+
     public async submitTransaction(transaction: Transaction<BigNumber>): Promise<TransactionReceipt> {
         if (!this.signer) throw new Error("Attempting to sign a transaction while not providing a signer");
         // TODO: figure out a way to propagate a warning up to the user in this scenario, we don't currently have a mechanism for error propagation, so will require infrastructure work
         // TODO: https://github.com/ethers-io/ethers.js/issues/321
         const tx = this.transactionToEthersTransaction(transaction);
         delete tx.from;
-        const receipt = await (await this.signer.sendTransaction(tx)).wait();
+        const txMetadataKey = `0x${transaction.data.substring(10)}`;
+        const txMetadata = this.transactionDataMetaData[txMetadataKey];
+        this.onTransactionStatusChanged(txMetadata, TransactionStatus.AWAITING_SIGNING);
+        const response = await this.signer.sendTransaction(tx);
```

Comment:
If this throws an exception we will have excess txMetadata hanging around. Would be good to handle if we could.

Reply:
will do